lib/common/socket/evloop.c.h

```diff
@@ -236,7 +314,7 @@ static void read_on_ready(struct st_h2o_evloop_socket_t *sock)
     if ((sock->_flags & H2O_SOCKET_FLAG_DONT_READ) != 0)
         goto Notify;
 
-    if ((err = on_read_core(sock->fd, sock->super.ssl == NULL ? &sock->super.input : &sock->super.ssl->input.encrypted)) != NULL)
+    if ((err = on_read_core(sock)) != NULL)
```

Comment:
I wonder if we could avoid the change of starting to pass `sock` to `on_read_core`.

Original code had the nice property of `on_read_core` being a freestanding function that could be evaluated without taking the rest of `st_h2o_evloop_socket_t` into consideration.

Perhaps what we want to consider is passing cmsg buffer as an optional argument to `on_read_core`, and call `handle_timestamp` from here after `on_read_core` returns.

Reply:
OK i made this change. It feels slightly odd in the code - I suppose an alternative might be adding an on_timestamp callback or something instead. Let me know what you think.